packages/filesystem/src/browser/file-dialog/file-dialog-service.ts

```diff
@@ -71,10 +71,10 @@ export class DefaultFileDialogService {
         return undefined;
     }
 
-    protected async getRootNode(folderToOpen?: FileStat): Promise<DirNode | undefined> {
+    protected async getDirNode(folderToOpen?: FileStat): Promise<DirNode | undefined> {
         const folder = folderToOpen || await this.fileSystem.getCurrentUserHome();
         if (folder) {
-            const rootUri = new URI(folder.uri).parent;
```

Comment:
we should still use a parent if someone passes `FileStat` with `false` as `isDirectory`

Reply:
You are right, but passing a file reference using a parameter named `folder` would be quite naughty!

Reply:
who knows what devs will do 😄